### PR TITLE
pbgopy: add service

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
### Description
A very simple service for [pbgopy](https://github.com/nakabonne/pbgopy).

This is my first contribution to home-manager, feedback is much appreciated :)

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/pbgopy.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+with lib;
+
+let
+  cfg = config.services.pbgopy;
+  package = pkgs.pbgopy;
+in {
+  meta.maintainers = [ maintainers.ivar ];
+
+  options.services.pbgopy = {
+    enable = mkEnableOption "Wether to enable the pbgopy.";
```
Comment (rycee) [inline modules/services/pbgopy.nix:12]:
```suggestion
    enable = mkEnableOption "pbgopy";
```

File: modules/services/pbgopy.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ package ];
+
+    systemd.user.services.pbgopy = {
+      Unit = {
+        Description = "pbgopy server for sharing the clipboard between devices";
+        After = [ "network.target" ];
```
Comment (rycee) [inline modules/services/pbgopy.nix:28]: There is no `network.target` in the user session
```suggestion
```

File: modules/services/pbgopy.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+let
+  cfg = config.services.pbgopy;
+  package = pkgs.pbgopy;
+in {
+  meta.maintainers = [ maintainers.ivar ];
+
+  options.services.pbgopy = {
+    enable = mkEnableOption "Wether to enable the pbgopy.";
+
+    ttlCache = mkOption {
```
Comment (rycee) [inline modules/services/pbgopy.nix:14]:
```suggestion
    cache.ttl = mkOption {
```

**Comment by rycee** [comment]
Thanks for the contribution, I've added a few comments 🙂

**Comment by IvarWithoutBones** [comment]
> Thanks for the contribution, I've added a few comments

Adressed all comments, thank you for the feedback.

**Comment by rycee** [comment]
Thanks! Merged to master.

**Pull request merged**